Pensel/firmware/applications/pensel_v2.c

```diff
@@ -86,108 +130,125 @@ void clear_critical_errors(void) {
+    int32_t time_until_next_cb_ms;
 
     // system configuration...
     HAL_Init();
     SystemClock_Config();
     configure_pins();
     clear_critical_errors();
 
-    retval = UART_init(250000);
-    check_retval_fatal(__FILE__, __LINE__, retval);
+    // Initialize UART
+    check_retval_fatal(__FILE__, __LINE__, UART_init(115200) );
 
     #ifdef WATCHDOG_ENABLE
-    if ( wdg_isSet() ) {
-        // set a report variable in critical errors
-        gCriticalErrors.wdg_reset = 1;
-        #ifdef WATCHDOG_CAPTURE
-            wdg_captureAlert();
-        #endif
-    }
-    retval = wdg_init();
-    check_retval_fatal(__FILE__, __LINE__, retval);
-    gPetWdg = true;
+        if ( wdg_isSet() ) {
+            // set a report variable in critical errors
+            gCriticalErrors.wdg_reset = 1;
+            #ifdef WATCHDOG_CAPTURE
+                wdg_captureAlert();
+            #endif
+        }
+        // Initialize the watchdog
+        check_retval_fatal(__FILE__, __LINE__, wdg_init() );
+        gPetWdg = true;
     #endif
 
     // peripheral configuration
-    retval = I2C_init();
-    check_retval_fatal(__FILE__, __LINE__, retval);
-
-    retval = orient_init();
-    check_retval_fatal(__FILE__, __LINE__, retval);
-
-    // Load Calibration
-    cal_loadFromFlash();
-    if (cal_checkValidity() != RET_OK) {
-        cal_loadDefaults();
-    }
+    // retval = I2C_init();
+    // check_retval_fatal(__FILE__, __LINE__, retval);
 
     LED_set(LED_0, 0);
     LED_set(LED_1, 0);
 
-    while (true) {
-        // Parse reports and such
-        // rpt_run();
+    MX_USB_PCD_Init();
+    USB_Init();
 
-        // get packets and such
-        LSM303DLHC_run();
+    uint32_t ledi = 0;
+    while (1)
+    {
 
-        // let the user know roughly how many workloops are ocurring
-        if (subcount == 100000) {
+        // toggle LED every 10,000 itterations over the workloop
+        if (ledi++ >= 10000) {
             LED_toggle(LED_1);
-            subcount = 0;
-        } else {
-            subcount++;
+            ledi = 0;
+        }
+        // UART_sendint(bDeviceState);
+        if (bDeviceState == CONFIGURED) {
+            CDC_Receive_DATA();
+            /*Check to see if we have data yet */
+            if (Receive_length  != 0) {
+                if (packet_sent == 1)
+                    CDC_Send_DATA ((unsigned char*)Receive_Buffer,Receive_length);
+                Receive_length = 0;
+            }
         }
+    }
 
-        // --- TODO: Check Mag Data ----
+    // initalize the scheduler and add some periodic tasks
+    scheduler_init(&main_schedule);
+    scheduler_add(&main_schedule, 0, heartbeat, &i);
+    scheduler_add(&main_schedule, 0, workloop_flash, &i);
+    scheduler_add(&main_schedule, 0, button_handler, &i);
+    #ifdef WATCHDOG_ENABLE
+        scheduler_add(&main_schedule, 0, watchdog_pet, &i);
+    #endif
 
+    UART_sendString("Initialized\r\n");
 
+    uint32_t sleep_until = 0;
+    while (true) {
```

Comment:
You are using `while (true)` and `while (1)` both within this same function, would be better to pick one format or the other simply for consistency